### ISSUE TITLE
Subset for webfonts

### ISSUE THREAD

**Issue by vincentbernat** [be5invis/Iosevka#238]
Hey!

For webfonts, a smaller file size would be welcome. What's the easiest way to only keep a subset of the characters? For example, I would be interested in those:
 - basic latin
 - latin-1
 - box drawing
 - box elements
 - private use (powerline)

**Comment by be5invis** [comment]
Actually I have some code (in Megaminx) can do sub-setting while keeping all the GSUB features...

**Comment by vincentbernat** [comment]
From [fonttools](https://github.com/fonttools/fonttools), there is `pyftsubset` to do that. I don't know if it enables to use subsets by names, but for my use, I do:

```
pyftsubset media/fonts/iosevka-term-regular.woff  --name-IDs+=0,4,6 --text-file=../glyphs.txt --flavor=woff --with-zopfli
pyftsubset media/fonts/iosevka-term-regular.woff2 --name-IDs+=0,4,6 --text-file=../glyphs.txt --flavor=woff2
```

With `glyphs.txt` a file containing all the glyphs I want to keep. Works fine for me.

**Comment by kaushalmodi** [comment]
@be5invis I'm looking forward to this too. At the moment, each variant (regular, italics, etc.) is taking up about 130KB.. totally to about 400KB of my 700KB site :)

**Comment by kaushalmodi** [comment]
@vincentbernat Oh, what a small world! Just today I was reading https://vincent.bernat.im/en/blog/2018-more-privacy-blog :)

I had a question.. how do you generate that glyphs.txt. Do you have a script that parses all code blocks in your site source to collect all required glyphs?

**Comment by vincentbernat** [comment]
I am using an extension for Python Markdown: https://github.com/vincentbernat/vincent.bernat.im/blob/master/extensions/markdown/glyphs.py.

**Comment by kaushalmodi** [comment]
Thanks. I am Hugo. So I will just create one manually and then append chars to it as needed. What does the glyphs.txt look like? Are the chars comma separated? Can you post your auto-generated version somewhere?

**Comment by vincentbernat** [comment]
```
 !"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_`abcdefghijklmnopqrstuvwxyz{|}~ ­ÀÉàçèéêôœ​‌… ←→↳⎜⎡⎣─│┊└├┬╯░▒♯❶❷❸❹❺❻❼❽
```

**Comment by kaushalmodi** [comment]
@vincentbernat Thank you! I finally got it working.. just that in the commands you listed in https://github.com/be5invis/Iosevka/issues/238#issuecomment-351527918, I needed to always use .ttf as inputs instead of .woff/woff2.

I am not a web/font dev, so then I verified that the generated subsets did actually have the intended glyphs using http://torinak.com/font/lsfont.html.

If I used woff/woff2 as inputs to `pyftsubset`, that lsfont utility failed to recognize the charmap.
